contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -12,37 +12,63 @@ import { IVault } from "../interfaces/IVault.sol";
 import { IOracle } from "../interfaces/IOracle.sol";
 import { IStrategy } from "../interfaces/IStrategy.sol";
 import { IUniswapV2Router } from "../interfaces/uniswap/IUniswapV2Router02.sol";
+import { IUniswapV3Router } from "../interfaces/uniswap/IUniswapV3Router.sol";
+import { IBalancerVault } from "../interfaces/balancer/IBalancerVault.sol";
+import { ICurvePool } from "../strategies/ICurvePool.sol";
 import "../utils/Helpers.sol";
 
 abstract contract BaseHarvester is Governable {
     using SafeERC20 for IERC20;
     using SafeMath for uint256;
     using StableMath for uint256;
 
-    event UniswapUpdated(address _address);
-    event SupportedStrategyUpdate(address _address, bool _isSupported);
+    enum SwapPlatform {
+        UniswapV2Compatible,
+        UniswapV3,
+        Balancer,
+        Curve
+    }
+
+    event SupportedStrategyUpdate(address strategyAddress, bool isSupported);
     event RewardTokenConfigUpdated(
-        address _tokenAddress,
-        uint16 _allowedSlippageBps,
-        uint16 _harvestRewardBps,
-        address _uniswapV2CompatibleAddr,
-        uint256 _liquidationLimit,
-        bool _doSwapRewardToken
+        address tokenAddress,
+        uint16 allowedSlippageBps,
+        uint16 harvestRewardBps,
+        SwapPlatform platform,
+        address swapRouterAddr,
+        bytes swapData,
+        uint256 liquidationLimit,
+        bool doSwapRewardToken
+    );
+    event RewardTokenSwapped(
+        address indexed rewardToken,
+        address indexed swappedInto,
+        SwapPlatform platform,
+        uint256 amountIn,
+        uint256 amountOut
+    );
+    event RewardProceedsTransferred(
+        address indexed token,
+        address farmer,
+        uint256 protocolShare,
```

Comment:
> one could expect this to represent a percentage of the total

Isn't that the case here? 😅 

Harvester swaps `x` token for `y` token and send a percentage of that (dictated by `harvestRewardBps`) to the farmer and the protocol? 

May be something like `farmerFee` (or `farmerReward`) and `protocolYield` would work?

Reply:
`farmerFee` and `protocolYield` sound great to me.

Sorry for not being clear enough with the previous message. I meant that when something is called `share` it could also be interpreted as being expressed in percentage points rather than part of the absolute number.